### PR TITLE
Clarify documentation for running a tor node

### PR DESCRIPTION
**Pull request by nathaniel-mahieu**
Previous wording suggested that no additional setup was required for a tor hidden service to be created.

Discussed here: #8145

### CODE REVIEW

File: doc/tor.md [hunk 1]
```diff
@@ -95,12 +95,18 @@ Starting with Tor version 0.2.7.1 it is possible, through Tor's control socket
...
 a connection to Tor can be made. It can be configured with the `-listenonion`,
 `-torcontrol` and `-torpassword` settings. To show verbose debugging
 information, pass `-debug=tor`.
+
+Connecting to Tor's control socket API requires one of two authentication methods to be 
+configured.  The most straightforward is the use of the `-torpassword` flag and a `hash-password` 
+which can be enabled in Tor configuration.  Alternatively the user running bitcoind must have 
+write access to the `CookieAuthFile` specified in Tor configuration.  This can be achieved 
+by adding both the user running tor, and the user running bitcoind to the same group and setting 
+permissions appropriately.
```
Comment (petertodd) [inline doc/tor.md]: Note that most distributions already have something along those lines, e.g. on Debian-based there is a debian-tor group, and you make cookie auth work by adding users who need it to the debian-tor group. I'd reword this to just say that you need to make sure the bitcoind user is part of the Tor group.

Anyone know how non-Debian distros do this?

Comment (laanwj) [inline doc/tor.md]: Yes, giving an example for Debian may help here.

File: doc/tor.md [hunk 2]
```diff
@@ -95,12 +95,18 @@ Starting with Tor version 0.2.7.1 it is possible, through Tor's control socket
...
+This means that if Tor is running (and proper authentication has been configured),
+Bitcoin Core automatically creates a hidden service to listen on. This will positively 
+affect the number of available .onion nodes.
 
 This new feature is enabled by default if Bitcoin Core is listening, and
 a connection to Tor can be made. It can be configured with the `-listenonion`,
 `-torcontrol` and `-torpassword` settings. To show verbose debugging
 information, pass `-debug=tor`.
+
+Connecting to Tor's control socket API requires one of two authentication methods to be
```
Comment (laanwj) [inline doc/tor.md]: I'd reword this a bit - cookie authentication is by far the most straightforward (even automatic) if bitcoin core has access to tor's cookie file. This is the case on Windows with Tor Browser Bundle, for example. If not, then extra work is needed to set it up.

File: doc/tor.md (hunk 1 above)
Comment (nathaniel-mahieu) [inline doc/tor.md]: Added the example for Debian. Are we worried that the debian-tor group includes too many permissions outside of this one cookie permission we are interested in?

File: doc/tor.md (hunk 2 above)
Comment (nathaniel-mahieu) [inline doc/tor.md]: Removed the opinion on simplicity.

File: doc/tor.md [hunk 3]
```diff
@@ -95,12 +95,22 @@ Starting with Tor version 0.2.7.1 it is possible, through Tor's control socket
...
 information, pass `-debug=tor`.
+
+Connecting to Tor's control socket API requires one of two authentication methods to be 
+configured.  For cookie authentication the user running bitcoind must have write access 
+to the `CookieAuthFile` specified in Tor configuration. In some cases this is 
+preconfigured and the creation of a hidden service is automatic.  If permission problems 
+are seen with `-debug=tor` they can be resolved by adding both the user running tor, and 
+the user running bitcoind to the same group and setting permissions appropriately.  On 
+debian based systems Tor automatically creates the debian-tor group and the user running 
+bitcoind can be added there to receive appropriate permissions. An alternative
```
Comment (petertodd) [inline doc/tor.md]: Rather than saying "Tor automatically creates" how about we just say "On Debian-based systems the user running bitcoind can be added to the debian-tor group, which has the appropriate permissions." - Debian's package management is what's automatically creating debian-tor, not Tor itself.

Comment (nathaniel-mahieu) [inline doc/tor.md]: Much better wording.

File: doc/tor.md [hunk 4]
```diff
@@ -95,12 +95,22 @@ Starting with Tor version 0.2.7.1 it is possible, through Tor's control socket
...
 `-torcontrol` and `-torpassword` settings. To show verbose debugging
 information, pass `-debug=tor`.
+
+Connecting to Tor's control socket API requires one of two authentication methods to be 
+configured.  For cookie authentication the user running bitcoind must have write access 
+to the `CookieAuthFile` specified in Tor configuration. In some cases this is 
+preconfigured and the creation of a hidden service is automatic.  If permission problems 
+are seen with `-debug=tor` they can be resolved by adding both the user running tor and 
+the user running bitcoind to the same group and setting permissions appropriately.  On 
+Debian-based systems the user running bitcoind can be added to the debian-tor group,
```
Comment (laanwj) [inline doc/tor.md]: No double spaces please :)

**Comment by petertodd** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/8203/commits/9e3ec74fac76f93bedaca882cc7053cc46199a61
